common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -292,6 +349,62 @@ static void putLong(long address, long value) {
         }
     }
 
+    static void putByte(byte[] data, int index, byte value) {
+        UNSAFE.putByte(data, ARRAY_BASE_OFFSET + index, value);
+    }
+
+    static void putShort(byte[] data, int index, short value) {
+        if (UNALIGNED) {
+            UNSAFE.putShort(data, ARRAY_BASE_OFFSET + index, value);
+        } else if (BIG_ENDIAN) {
+            putByte(data, index, (byte) (value >>> 8));
+            putByte(data, index + 1, (byte) value);
+        } else {
+            putByte(data, index + 1, (byte) (value >>> 8));
+            putByte(data, index, (byte) value);
+        }
+    }
+
+    static void putInt(byte[] data, int index, int value) {
+        if (UNALIGNED) {
+            UNSAFE.putInt(data, ARRAY_BASE_OFFSET + index, value);
```

Comment:
No as we reverse these if wee need to here:
https://github.com/netty/netty/pull/4361/files#diff-97f49e73448689db365493e3a0f77824R63